src/features/user/components/ChangeInfoForm/ChangeInfoForm.tsx

```diff
@@ -0,0 +1,87 @@
+import { useAppDispatch, useAppSelector } from "@/app/hooks"
+import FormikTextField from "@/components/FormikTextField/FormikTextField"
+import changeUserInfoSchema, {
+  ChangeUserInfoFormValues,
+} from "@/libs/yup/schemas/changeUserInfo"
+import Grid from "@mui/material/Grid/Grid"
+import { Form, Formik } from "formik"
+import { changeUserData } from "../../userThunk"
+import SubmitFormButton from "../../../../components/SubmitFormButton/SubmitFormButton"
+import ResetFormButton from "../../../../components/ResetFormButton/ResetFormButton"
+import { useTranslation } from "react-i18next"
+
+const ChangeInfoForm = () => {
+  const dispatch = useAppDispatch()
+  const { t } = useTranslation(["forms"])
+  const {
+    userData: { firstName, lastName, email },
+  } = useAppSelector((store) => store.user)
+  const initialValues: ChangeUserInfoFormValues = {
+    firstName,
+    lastName,
+    email,
+  }
+  return (
+    <Formik
+      onSubmit={async (values) => {
+        await dispatch(changeUserData(values))
+      }}
+      initialValues={initialValues}
+      validationSchema={changeUserInfoSchema}
+    >
+      {({ isSubmitting, values, setValues }) => {
+        // in this case I had to create my own 'dirty' and 'resetForm' because after user submitted new values, formik's: 'dirty' and 'resetForm' were still using the old values
+        const dirty =
```

Comment:
Instead of creating your own dirty and resetForm logic, just add enableReinitialize prop to the formik settings